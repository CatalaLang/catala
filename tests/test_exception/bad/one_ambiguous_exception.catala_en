## Test

```catala
declaration scope A:
  context x content integer
  context y content integer

scope A:
  definition x equals 0

  exception
  definition x equals 1

  definition y equals 2

  definition y equals 4

  exception
  definition y equals 3
```

```catala-test-inline
$ catala Interpret -s A
[ERROR]
This exception can refer to several definitions. Try using labels to disambiguate

Ambiguous exception
┌─⯈ tests/test_exception/bad/one_ambiguous_exception.catala_en:18.3-19.15:
└──┐
18 │   exception
   │   ‾‾‾‾‾‾‾‾‾
19 │   definition y equals 3
   │   ‾‾‾‾‾‾‾‾‾‾‾‾
   └─ Test

Candidate definition
┌─⯈ tests/test_exception/bad/one_ambiguous_exception.catala_en:16.14-16.15:
└──┐
16 │   definition y equals 4
   │              ‾
   └─ Test

Candidate definition
┌─⯈ tests/test_exception/bad/one_ambiguous_exception.catala_en:14.14-14.15:
└──┐
14 │   definition y equals 2
   │              ‾
   └─ Test
#return code 123#
```
