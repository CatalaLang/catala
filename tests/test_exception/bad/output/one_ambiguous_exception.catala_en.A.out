[ERROR] This exception can refer to several definitions. Try using labels to disambiguate
[ERROR] 
[ERROR] Ambiguous exception
[ERROR]   --> test_exception/bad/one_ambiguous_exception.catala_en
[ERROR]    | 
[ERROR] 16 |   definition y equals 4
[ERROR]    |                       
[ERROR] 17 | 
[ERROR]    | 
[ERROR] 18 |   exception
[ERROR]    |   ^^^^^^^^^
[ERROR] 19 |   definition y equals 3
[ERROR]    |   ^^^^^^^^^^^^^^^^^^^^^^
[ERROR]    + Test
[ERROR] 
[ERROR] Candidate definition
[ERROR]   --> test_exception/bad/one_ambiguous_exception.catala_en
[ERROR]    | 
[ERROR] 16 |   definition y equals 4
[ERROR]    |              ^
[ERROR]    + Test
[ERROR] 
[ERROR] Candidate definition
[ERROR]   --> test_exception/bad/one_ambiguous_exception.catala_en
[ERROR]    | 
[ERROR] 14 |   definition y equals 2
[ERROR]    |              ^
[ERROR]    + Test
